tests/problems/space/test_alignment_problem.py

```diff
@@ -122,49 +123,51 @@ def test_solve_unbalanced(self, adata_space_rotate: AnnData):
-
-    #     dfs = []
-    #     print(unique_batches)
-    #     for i in range(len(unique_batches)-1):
-    #         batch1 = unique_batches[i]
-    #         batch2 = unique_batches[i + 1]
-    #         indices = np.where(
-    #           (adata_space_rotate.obs[batch_column] == batch1) | (adata_space_rotate.obs[batch_column] == batch2)
-    #           )[0]
-    #         adata_subset = adata_space_rotate[indices]
-    #         sc.pp.neighbors(adata_subset, n_neighbors=15, use_rep="X_pca")
-    #         dfs.append(
-    #             pd.DataFrame(
-    #                 index=adata_subset.obs_names,
-    #                 columns=adata_subset.obs_names,
-    #                 data=adata_subset.obsp["connectivities"].A.astype("float64"),
-    #             )
-    #         )
-
-    #     ap: AlignmentProblem = AlignmentProblem(adata=adata_space_rotate)
-    #     ap = ap.prepare(batch_key=batch_column, joint_attr={"attr": "obsm", "key": "X_pca"})
-
-    #     ap[('0', "1")].set_graph_xy(dfs[0], cost="geodesic")
-    #     ap = ap.solve(max_iterations=2, lse_mode=False)
-
-    #     ta = ap[('0', "1")].xy
-    #     assert isinstance(ta, TaggedArray)
-    #     assert isinstance(ta.data_src, np.ndarray)  # this will change once OTT-JAX allows for sparse matrices
-    #     assert ta.data_tgt is None
-    #     assert ta.tag == Tag.KERNEL
-    #     assert ta.cost == "geodesic"
-
-    #     ta = ap[('0', "1")].xy
-    #     assert isinstance(ta, TaggedArray)
-    #     assert isinstance(ta.data_src, np.ndarray)  # this will change once OTT-JAX allows for sparse matrices
-    #     assert ta.data_tgt is None
-    #     assert ta.tag == Tag.KERNEL
-    #     assert ta.cost == "geodesic"
+    @pytest.mark.parametrize("key", ["connectivities", "distances"])
+    def test_geodesic_cost(self, adata_space_rotate: AnnData, key: str):
+        import pandas as pd
+
+        import scanpy as sc
+
+        batch_column = "batch"
+        unique_batches = adata_space_rotate.obs[batch_column].unique()
+
+        dfs = []
+        for i in range(len(unique_batches) - 1):
+            batch1 = unique_batches[i]
+            batch2 = unique_batches[i + 1]
+            indices = np.where(
+                (adata_space_rotate.obs[batch_column] == batch1) | (adata_space_rotate.obs[batch_column] == batch2)
+            )[0]
+            adata_subset = adata_space_rotate[indices]
+            sc.pp.neighbors(adata_subset, n_neighbors=15, use_rep="X_pca")
+            dfs.append(
+                pd.DataFrame(
+                    index=adata_subset.obs_names,
+                    columns=adata_subset.obs_names,
+                    data=adata_subset.obsp["connectivities"].A.astype("float64"),
+                )
+            )
+
+        ap: AlignmentProblem = AlignmentProblem(adata=adata_space_rotate)
+        ap = ap.prepare(batch_key=batch_column, joint_attr={"attr": "obsm", "key": "X_pca"})
+
+        ap[("0", "1")].set_graph_xy(dfs[0], cost="geodesic")
```

Comment:
@giovp why does this work now? I thought ott-jax cannot handle geodesic in FGW?

Reply:
it's handled here as you suggested https://github.com/theislab/moscot/blob/f38ea663b2887b94878e740072d5ebaad0fbd6d8/src/moscot/backends/ott/solver.py#L105-L107

Reply:
ok great!